test/e2e/framework/framework.go

```diff
@@ -0,0 +1,139 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/client-go/1.5/kubernetes"
+	"k8s.io/client-go/1.5/pkg/api"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	"k8s.io/client-go/1.5/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/1.5/pkg/labels"
+	"k8s.io/client-go/1.5/pkg/util/yaml"
+	"k8s.io/client-go/1.5/tools/clientcmd"
+)
+
+var Global *Framework
+
+type Framework struct {
+	KubeClient  kubernetes.Interface
+	HTTPClient  *http.Client
+	MasterHost  string
+	Namespace   *v1.Namespace
+	OperatorPod *v1.Pod
+}
+
+// Setup setups a test framework and points "Global" to it.
+func Setup() error {
+	kubeconfig := flag.String("kubeconfig", "", "kube config path, e.g. $HOME/.kube/config")
+	opImage := flag.String("operator-image", "", "operator image, e.g. quay.io/coreos/prometheus-operator")
+	ns := flag.String("namespace", "prometheus-operator-e2e-tests", "e2e test namespace")
+	flag.Parse()
+
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	cli, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	httpc := cli.Core().GetRESTClient().Client
+	if err != nil {
+		return err
+	}
+
+	namespace, err := cli.Core().Namespaces().Create(&v1.Namespace{
+		ObjectMeta: v1.ObjectMeta{
+			Name: *ns,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	Global = &Framework{
+		MasterHost: config.Host,
+		KubeClient: cli,
+		HTTPClient: httpc,
+		Namespace:  namespace,
+	}
+	return Global.setup(*opImage)
+}
+
+func (f *Framework) setup(opImage string) error {
+	fmt.Println("setting up e2e tests ...")
+	if err := f.setupPrometheusOperator(opImage); err != nil {
+		fmt.Printf("failed to setup prometheus operator: %v\n", err)
+		return err
+	}
+	fmt.Println("e2e setup successfully")
+	return nil
+}
+
+func (f *Framework) setupPrometheusOperator(opImage string) error {
+	fn, err := filepath.Abs("../../deployment.yaml")
+	if err != nil {
+		return err
+	}
+
+	deployManifest, err := os.Open(fn)
+	if err != nil {
+		return err
+	}
+
+	deploy := v1beta1.Deployment{}
+	err = yaml.NewYAMLOrJSONDecoder(deployManifest, 100).Decode(&deploy)
+	if err != nil {
+		return err
+	}
+
+	err = CreateDeployment(f.KubeClient, f.Namespace.Name, &deploy)
+	if err != nil {
+		return err
+	}
+
+	opts := api.ListOptions{LabelSelector: labels.SelectorFromSet(deploy.Spec.Template.ObjectMeta.Labels)}
+	pl, err := WaitForPodsReady(f.KubeClient, f.Namespace.Name, 60*time.Second, 1, opts)
+	if err != nil {
+		return err
+	}
+	f.OperatorPod = &pl.Items[0]
+
+	err = WaitPrometheusTPRReady(f.HTTPClient, 5*time.Second, 60*time.Second, f.MasterHost, f.Namespace.Name)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func Teardown() error {
```

Comment:
That would be a method to `Framework` then.
The logging should be moved into the caller.